tracim/views/core_api/schemas.py

```diff
@@ -355,3 +382,129 @@ class ContentDigestSchema(marshmallow.Schema):
                     'for sub-contents. Default is True. '
                     'In first version of the API, this field is always True',
     )
+
+
+#####
+# Content
+#####
+
+class ContentSchema(ContentDigestSchema):
+    current_revision_id = marshmallow.fields.Int(example=12)
+    created = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='Content creation date',
+    )
+    author = marshmallow.fields.Nested(UserDigestSchema)
+    modified = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='date of last modification of content',
+    )
+    last_modifier = marshmallow.fields.Nested(UserDigestSchema)
+
+
+class ThreadContentSchema(ContentSchema):
+    content_type = marshmallow.fields.Str(
+        example='thread',
+        validate=OneOf([content.slug for content in CONTENT_DEFAULT_TYPE]),
+    )
+    raw_content = marshmallow.fields.String('Description of Thread')
+
+
+class HtmlDocumentContentSchema(ContentSchema):
+    content_type = marshmallow.fields.Str(
+        example='html-content',
+        validate=OneOf([content.slug for content in CONTENT_DEFAULT_TYPE]),
+    )
+    raw_content = marshmallow.fields.String('<p>Html page Content !</p>')
+
+#####
+# Revision
+#####
+
+
+class RevisionSchema(ContentDigestSchema):
+    comments_ids = marshmallow.fields.List(marshmallow.fields.Int(example=4))
+    revision_id = marshmallow.fields.Int(example=12)
+    created = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='Content creation date',
+    )
+    author = marshmallow.fields.Nested(UserDigestSchema)
+
+
+class ThreadRevisionSchema(RevisionSchema):
```

Comment:
same remark. Define a unique `TextBasedContentRevisionSchema` used in place of `ThreadRevisionSchema` and `HtmlDocumentRevisionSchema`

Reply:
https://github.com/tracim/tracim/issues/618